cypress/integration/formr-b/formr-b.spec.ts

```diff
@@ -119,13 +133,115 @@ describe("Form R (Part B)", () => {
+    ).click();
+
+    // -------- Section 3 Declarations relating to Good Medical Practice -----------
+    cy.get("[data-cy=legendFieldset3]").should("include.text", "Section 3");
+    cy.get("[data-cy=mainWarning3]").should("be.visible");
+    cy.get("[data-cy=declarations]").should("be.visible");
+    cy.get("[data-cy=healthStatement]").should("be.visible");
+    // Navigate back to section 2
+    cy.get(
+      ".nhsuk-pagination-item--previous > .nhsuk-pagination__link > .nhsuk-pagination__page"
+    ).click();
+    cy.get("[data-cy=mainWarning2]").should("be.visible");
+    // Section 2 data is unchanged
+    cy.get("#totalLeave").should("be.visible").should("contain.value", "11");
+    // Return to section 3
+    cy.get(
+      ".nhsuk-pagination-item--next > .nhsuk-pagination__link > .nhsuk-pagination__page"
+    ).click();
+    cy.get("[data-cy=isHonest0]")
+      .should("be.visible")
+      .should("contain.value", "")
+      .click();
+    cy.get("[data-cy=isHealthy0]")
+      .should("be.visible")
+      .should("contain.value", "")
+      .click();
+    cy.get("[data-cy=isComplying0]").should("not.be.visible");
+    cy.get("[data-cy=isWarned0]")
+      .should("be.visible")
+      .should("contain.value", "")
+      .click();
+    cy.get("[data-cy=isComplying0]").should("be.visible").click();
+    cy.get("[data-cy=healthStatementTextInput]")
+      .should("be.visible")
+      .type("I'm in astonishingly excellent health.");
+
+    // Go to view / edit / submit
+    cy.get(
+      ".nhsuk-pagination-item--next > .nhsuk-pagination__link > .nhsuk-pagination__page"
+    ).click();
+    cy.get("[data-cy=BtnEditSection1]").should("be.visible");
+    cy.get("[data-cy=gmcNumber]").should("be.visible");
+    cy.get("[data-cy=BtnEditSection2]").should("be.visible");
+    cy.get("[data-cy=typeOfWork3]").should("be.visible");
+    cy.get("[data-cy=BtnEditSection3]").should("be.visible");
+
+    // Click edit btn to edit section 1 details
+    cy.get(".nhsuk-back-link__link").should("be.visible");
+    cy.get("[data-cy=BtnEditSection1]").should("be.visible").click();
+    cy.get("#gmcNumber").clear().type("22222222");
+    cy.get(".nhsuk-pagination__page").click();
+
+    cy.get("[data-cy=workTypeInput2]").should(
+      "contain.value",
+      "Another type of work"
+    );
+    cy.get(
+      ".nhsuk-pagination-item--next > .nhsuk-pagination__link > .nhsuk-pagination__page"
+    ).click();
+
+    cy.get("[data-cy=healthStatementTextInput]").should(
+      "contain.value",
+      "I'm in astonishingly excellent health."
+    );
+    cy.get(
+      ".nhsuk-pagination-item--next > .nhsuk-pagination__link > .nhsuk-pagination__page"
+    ).click();
+    cy.get("[data-cy=gmcNumber]").should("have.text", "22222222");
+
+    cy.get(':nth-child(2) > [type="submit"]').should("be.visible").click();
+
+    // See list of saved forms
+    cy.contains("Submitted forms").should("be.visible");
+    cy.get("[data-cy=submittedForm0]").should("be.visible");
```

Comment:
May be hypothetical based on our approach for tests - what if we have some saved forms already? Doesn't this check for the first row which might not be the current saved form?

Reply:
Yeah, thinking was based on a testing env where only one saved form would exist. But in the meantime it will, as you say, check to see if there are any saved forms (there should be at least one) by getting the first in the list.